gov/politeia/types/types.go

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2019, The Decred developers
+// See LICENSE for details.
+
+package types
+
+import piapi "github.com/decred/politeia/politeiawww/api/v1"
+
+// ProposalInfo holds the proposal details as document here
+// https://github.com/decred/politeia/blob/master/politeiawww/api/v1/api.md#proposal.
+// It also holds the votes status details.
+type ProposalInfo struct {
+	ID              int                `json:"id" storm:"id,increment"`
```

Comment:
I found it problematic to query a nested field so that is why I used an ID field to help query the proposals. Also the field is managed by the db, so very few chances of a duplicate db entry for the same proposal being updated exist.

Reply:
To be clear, I'm not talking about querying a nested field. The ID field itself can be a string set to the token, and you'd only have one unique id for the proposal instead of two. It likely would never be an issue unless there was a problem coming from the Politeia API though, so if you're committed to the auto-increment, it's probably okay.